front/components/assistant_builder/AssistantBuilderDataSourceModal.tsx

```diff
@@ -211,98 +40,39 @@ export default function AssistantBuilderDataSourceModal({
       onClose={() => {
-        // If we are in the PickKind mode, we close the modal, otherwise we go back to the PickKind mode
-        if (displayMode === "PickKind") {
-          setOpen(false);
-        }
-        // We are one of the selection modes, we didn't save so we cancel the changes and go back to the PickKind mode
-        else {
-          setDataSourceConfigurations(initialDataSourceConfigurations);
-          setDisplayMode("PickKind");
-        }
+        setSelectionConfigurations(initialDataSourceConfigurations);
+        setOpen(false);
       }}
       onSave={() => {
-        onSave(dataSourceConfigurations);
+        onSave(selectionConfigurations);
         setOpen(false);
       }}
-      hasChanged={displayMode !== "PickKind"}
-      variant="full-screen"
+      hasChanged={hasChanged}
+      variant="side-md"
       title="Manage data sources selection"
     >
       <div className="w-full pt-12">
-        {displayMode === "PickKind" && (
-          <PickDataSource
-            onPick={(dsView) => {
-              setSelectedDataSourceView(dsView);
-              setDisplayMode("SelectFromManaged");
-            }}
-            onPickFolders={() => {
-              setDisplayMode("SelectFromFolder");
-            }}
-            onPickWebsites={() => {
-              setDisplayMode("SelectFromWebsite");
-            }}
-          />
-        )}
-
-        {displayMode === "SelectFromFolder" && (
-          <FolderOrWebsiteResourceSelector
-            owner={owner}
-            type="folder"
-            selectedNodes={alreadySelectedFolders}
-            onSelectChange={onSelectChangeFolderOrWebsite}
-          />
-        )}
-
-        {displayMode === "SelectFromWebsite" && (
-          <FolderOrWebsiteResourceSelector
-            owner={owner}
-            type="website"
-            selectedNodes={alreadySelectedWebsites}
-            onSelectChange={onSelectChangeFolderOrWebsite}
-          />
-        )}
-
-        {displayMode === "SelectFromManaged" && selectedDataSourceView && (
-          <DataSourceViewResourceSelector
-            dataSourceView={selectedDataSourceView}
-            owner={owner}
-            selectedResources={
-              getConfiguration(dataSourceConfigurations, selectedDataSourceView)
-                ?.selectedResources ?? []
-            }
-            isSelectAll={
-              getConfiguration(dataSourceConfigurations, selectedDataSourceView)
-                ?.isSelectAll ?? false
-            }
-            onSelectChange={onSelectChangeManaged}
-            toggleSelectAll={onToggleSelectAll}
-          />
+        {orderDatasourceViewByImportance(dataSourceViews).map(
+          (dataSourceView) => (
+            <DataSourceViewSelector
+              key={dataSourceView.sId}
+              owner={owner}
+              selectionConfiguration={
+                selectionConfigurations[dataSourceView.sId] ??
+                defaultSelectionConfiguration(dataSourceView)
```

Comment:
Not sure about the return value of this function, but I assume it returns a "new" value on every call? If so, we might want to memoize.

Reply:
i don't think it's worth memoizing as it will be added to the `selectionConfigurations` and next time you'll not call it.